src/layers/core/choropleth-layer/choropleth-layer.js

```diff
@@ -210,3 +207,89 @@ export default class ChoroplethLayer extends BaseLayer {
   }
 
 }
+
+// convert geojson to choropleths
+// @param {object} data - geojson object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function extractChoropleths(data) {
+  const normalizedGeojson = normalize(data);
+  const result = [];
+
+  normalizedGeojson.features.map((feature, featureIndex) => {
+    const choropleths = featureToChoropleths(feature);
+    choropleths.forEach(choropleth => {
+      choropleth.featureIndex = featureIndex;
+    });
+    result.push(...choropleths);
+  });
+  return result;
+}
+
+// normalize the coordinates array of a geojson feature
+// @param {object} feature - geojson feature object
+// @returns {array.array.array.point} array of geojson polygon-like coordinates
+function featureToChoropleths(feature) {
+  const {coordinates, type} = feature.geometry;
+  let choropleths;
+
+  switch(type) {
+  case 'MultiPolygon':
+    choropleths = coordinates;
+    break;
+  case 'Polygon':
+    choropleths = [coordinates];
+    break;
+  case 'LineString':
```

Comment:
@ibgreen I think LineStrings are out of scope for this component - for now should def do a layer for each primitive. Would love a robust and generic GeoJSON layer that can render anything in the spec - basically trying to hack one together right now, can contribute code from that if I can split it out